main.js

```diff
@@ -0,0 +1,362 @@
+// Javascript 
+
+// array variable to contain all posts
+var posts = [];
+
+// vars for HTML elements
+var pageView = document.getElementsByClassName('post-comment-thread')[0];
+
+var postInput = document.getElementsByClassName('post-text')[0];
+var postUserName = document.getElementsByClassName('post-name')[0];
+var postBtn = document.getElementsByClassName('post-btn')[0];
+
+// MessageBoard Class
+class MessageBoard {
+  constructor(currentMessageBoard, posts){
+    this.currentMessageBoard = currentMessageBoard;
+    this.posts = posts;
+  }
+
+  // returns current posts of MessageBoard instance
+  getPosts() {
+    return this.posts;
+  }
+  setPosts(posts){
+    this.posts = posts;
+  }
+  getPost(postId){
+    var post;
+    for(var i = 0; i < this.getPosts().length; i++){
+      post = this.getPosts()[i];
+      if(post === this.getPosts()[postId]){
+         isPost = true;
+      }
+    }
+    return isPost;
+  }
+
+  // function to set DOM elements to browser  
+  renderPage() {
+    // clears html DOM element
+    var threadBody = document.getElementsByClassName('thread-body')[0];
+    while(threadBody.hasChildNodes()){
+      threadBody.removeChild(threadBody.firstChild);
+    }
+    /* prevents displaying 'undefined' text when page is first loaded and has no posts and then appends the DOM elements via a call to renderPosts() function on the MessageBoard class instance variable */
+    if(!this.renderPosts() === undefined){
+      threadBody.append(this.renderPosts());
+      for(var i = 0; i < this.getPosts().length; i++){
+        if(this.getPosts()[i].getComments().length !== 0){
+          this.getPosts()[i].renderComments();
+        }
+      }
+    }
+  }
+
+  // checks input values for Post submissions and creates/pushes new Post instance to global variable posts[] array
+  addPost() {
+    switch(true){
+      case(postInput.value === ""):
+        alert("'Post Text' input cannot be blank.");
+        break;
+      case(postUserName.value === ""):
+        alert("'Your Name' input cannot be blank.");
+        break;
+      default:
+        // creates new Post class instance with and sets values for attributes then pushes instance of Post to posts[] array
+        var comments = [];
+        var newPost = new Post();
+        newPost.post = postInput.value;
+        newPost.postUserName = postUserName.value;
+        newPost.comments = comments;
+        newPost.postId = posts.length;
+        posts.push(newPost); 
+        this.setPosts(posts);
+    }
+  }
+
+  /**  
+    method to delete post instance in DOM based on target - called in onclick method set in buildPostDOM() method

```

Comment:
I like your comments - would be good to put types of your params as well using https://jsdoc.app/ jsdoc as a standard